src/superannotate/lib/app/interface/base_interface.py

```diff
@@ -1,14 +1,140 @@
-from lib import get_default_controller
-from lib.infrastructure.repositories import ConfigRepository
+import functools
+import sys
+from abc import ABC
+from abc import abstractmethod
+from inspect import signature
+from types import FunctionType
+from typing import Iterable
+from typing import Sized
 
+from lib.app.helpers import extract_project_folder
+from lib.core.reporter import Session
+from mixpanel import Mixpanel
+from version import __version__
 
-class BaseInterfaceFacade:
-    def __init__(self):
-        self._config_path = None
-        self._controller = get_default_controller()
 
+class BaseInterfaceFacade(ABC):
     @property
-    def controller(self):
-        if not ConfigRepository().get_one("token"):
-            raise Exception("Config does not exists!")
-        return self._controller
+    @abstractmethod
+    def host(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def token(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def logger(self):
+        raise NotImplementedError
+
+
+class Tracker:
+    TEAM_DATA = None
+    INITIAL_EVENT = {"event_name": "SDK init", "properties": {}}
+    INITIAL_LOGGED = False
+
+    @staticmethod
+    def get_mp_instance() -> Mixpanel:
+        # if "api.annotate.online" in get_default_controller()._backend_url:
+        #     return Mixpanel("ca95ed96f80e8ec3be791e2d3097cf51")
+        return Mixpanel("e741d4863e7e05b1a45833d01865ef0d")
+
+    @staticmethod
+    def get_default_payload(team_name, user_id, project_name=None):
+        return {
+            "SDK": True,
+            "Paid": True,
+            "Team": team_name,
+            "Team Owner": user_id,
+            "Project Name": project_name,
+            "Project Role": "Admin",
+            "Version": __version__,
+        }
+
+    def __init__(self, function):
+        self.function = function
+        functools.update_wrapper(self, function)
+
+    @staticmethod
+    def extract_arguments(function, *args, **kwargs) -> dict:
+        bound_arguments = signature(function).bind(*args, **kwargs)
+        bound_arguments.apply_defaults()
+        return dict(bound_arguments.arguments)
+
+    @staticmethod
+    def default_parser(function_name: str, kwargs: dict) -> tuple:
+        properties = {}
+        for key, value in kwargs.items():
+            if key == "self":
+                continue
+            elif key == "project":
+                properties["project_name"], properties["folder_name"] = extract_project_folder(value)
+            elif isinstance(value, str) and key == "project":
```

Comment:
Line 75 is unreachable. if `key == "project"` the `elif` will not be triggered. if `key != "project"` the `elif` will always return `False`.